test/fixtures/iotFleetProvisioning/registerDevice.js

```diff
@@ -0,0 +1,148 @@
+'use strict';
+// NOTE: `aws-iot-device-sdk-v2` is bundled into the deployment package
+// eslint-disable-next-line import/no-unresolved
+const { mqtt, io, iot, iotidentity } = require('aws-iot-device-sdk-v2');
+
+const createDeferredMessageResolver = () => {
+  let resolve;
+  let reject;
+  const resolvedMessage = new Promise((promiseResolve, promiseReject) => {
```

Comment:
Using term _resolved_ for promise object which is _not resolved_ it's confusing.

Best name for this property is probably `promise` (and in this case, generic `deferred` interface is exposed, context in which this function is used should not influence internal naming)